src/adm.rs

```diff
@@ -123,7 +112,36 @@ impl AdmFilter {
         tile: &mut AdmTile,
         tags: &mut Tags,
     ) -> HandlerResult<()> {
-        check_url(&tile.click_url, "Click", &filter.click_hosts, tags)
+        let url = &tile.click_url;
+        let species = "Click";
+        tags.add_tag("type", species);
+        tags.add_extra("url", &url);
+        // Check the required fields are present for the `click_url`
+        // pg 15 of 5.7.21 spec.
+        let mut check_keys = vec![
+            "ctag".to_owned(),
+            "version".to_owned(),
+            "key".to_owned(),
+            "ci".to_owned(),
+            "aespFlag".to_owned(),
+        ];
+        check_keys.sort();
```

Comment:
Looks like the latest API doc has one more probably new addition here, an optional "click-status" field that we should allow if present. It's only for clicks.

Reply:
ok. I'll add a required and optional param check.